code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -309,12 +310,12 @@ var/global/list/obj/item/device/pda/PDAs = list()
 /obj/item/device/pda/pickup(mob/user)
 	if(fon)
 		set_light(0)
-		user.set_light(user.light_range + f_lum)
+		user.set_light(user.light_range + f_lum, user.light_power + f_pow)

```

Comment:
Check out flashlight code for this. Setting the user light level in this way is prone to failure.